packages/components/src/components/forms/input/CheckboxInput.tsx

```diff
@@ -65,7 +64,19 @@ class CheckboxInputImpl extends DisableableInput<CheckboxInputProps, CheckboxInp
         });
         if (this.props.formsy && Utils.isFunction(this.props.setValue))
             this.props.setValue(checked);
-    }
+    };
+
+    toggleDisabled = () => {
+        const { value } = this.props;
+        const { checked } = this.state;
+
+        this.setState(() => {
+            return {
+                isDisabled: !this.state.isDisabled,
+                checked: this.state.isDisabled ? checked : (value === true || value === "true"),
```

Comment:
Instead of using this.state here inside of the setState callback, I think you want to use the "current" version of the state variable which would be passed as the first param:
`this.setState((state) => ({isDisabled: !state.isDisabled, ...}))`